docs/testplan/PFC_Congestion_Oversubscription_Test_Plan.md

```diff
@@ -0,0 +1,318 @@
+PFC pause frames should automatically be generated by the device under test (DUT) to throttle back the lossless traffic sources causing the over-subscription, avoiding any lossless traffic drops.
+The two aspects that will be verified are whether:
+*	Packet drops really happen when lossy traffic suffers from congestion without triggering PFC
+*	WRR scheduler works as expected
+
+#### Background
+This test uses a similar diagram and description of the multi-ASIC router as the ‘Many-to-One Lossless Traffic Congestion’ test case.
+In this variant, the egress port packet forwarding engine, facing similar oversubscription demands as the previous test, will fairly grant resources to the VOQs in the packet forwarding engines.  The ingress port VOQs, with not enough allocation granted for the lossy traffic streams causing the congestion, will eventually run out of the memory buffer resources for the aggressor VOQs and will be forced drop some of their frames.
+
+#### Testbed Setup
+The test bed setup consists of three IXIA ports and a SONiC router as the device under test (DUT). All IXIA ports should have the same line speed.  The three DUT ports will be chosen at random across all the available ports in the router in order to eventually cover combinations of ports within the same line card and across line cards (or ports within the same ASIC vs. different ASICs on a single-card, multi-ASIC design).
+
+<p float="left">
+  <img src="Img/PFC_Testcase2.png" width="500"  hspace="250"/>
+</p>
+
+
+#### Test steps
+In this experiment we will create a total of four streams, with each of the two test ports sending two streams each. The aggregate bandwidth will be slightly over 100%:
+  *	The background traffic will include two lossless traffic streams, with the SONiC default lossless priorities of 3 and 4, and each having a 25% bandwidth.
+  *	The test data traffic will consist of two lossy traffic streams, each with randomly chosen priorities (0..2, 5..7), and randomized so that either both streams are at 30% bandwidth each, or for one stream to be at 25% while the other is at 30% bandwidth.
+
+The mapping of background and test data traffic streams to IXIA transmit ports can be randomized while keeping two streams per port; for example, one test port could send two lossless traffic streams and the second port two lossy streams, or each port could send two mixed streams, one lossless, one lossy.
+
+This experiment needs the following eight steps:
+1.	Start the background traffic from IXIA Tx ports 1 & 2; the combined line rate will be 50% of the egress port capacity (25% line rate bandwidth per stream)
+2.	Start the test traffic from IXIA Tx ports 1 & 2; the additional increase in bandwidth will be either 55% or 60% depending on randomization (one stream with 25% BW, second stream with 30%, or both at 30% each), for a total oversubscription of either 105% or 110%.

```

Comment:
what is the frame size that is used in ixia?

Reply:
By default we are using 1024 FS